### ISSUE TITLE
Battle Armor cannot be replaced

### ISSUE THREAD

**Issue by Tecmes** [MegaMek/mekhq#2675]
[49.2+410]

AS seen in the screenshot, this is Corona heavy BA unit. Four armors are destroyed. I can't repair them nor their armor, because the repair tool seems to expect "BA standard (Basic)" armors (or something), eventhough MHQ does lists the Corona heavy as parts. Even if I procure those parts, I still can't repair, not even with GM>Complete repair.
The only solution was to "Restore Unit".
![unrepairable_BA](https://user-images.githubusercontent.com/80174634/120371746-3dcb5580-c2e4-11eb-9dc2-0bb1e3cdfa26.png)

**Comment by Tecmes** [comment]
Also when trying to reload BA SRM, it says it works, but nothing happens. If using Instant Repair, itries with every tech, filling the log with messages and blocking the software for a couple seconds. I *think* that it's trying to reload SRMs on destroyed suits.

**Comment by Tecmes** [comment]
Another one... **When replacing a missing suit for a BA unit when there is no available pilots to "fill" them**, it "fills" it with a ghost. In the Hangar, the TRO entry seems A-OK. However
1) the entry is Purple, which would indicate a missing personel
2) if you check the assigned pilots, there's not the full complement.
3) In the TO&E, it appears weird... the TRO part is fine, but it says also "No crew"...

**Comment by Tecmes** [comment]
Another related issue: when a BA pilot retires, it brings its whole unit with it - all 4 or 5 suits...

**Comment by Tecmes** [comment]
Also... in the TO&E view, the list of crew (horizontal strip on top of the right pane), the crew are listed as vehicule crew (as Drivers).

**Comment by binaryspica** [comment]
> Also... in the TO&E view, the list of crew (horizontal strip on top of the right pane), the crew are listed as vehicule crew (as Drivers).

I don't think that it is a problem; each of them are actually the individual driver of a battle armor. Also it simplify the issue as well.

**Comment by gsparks3** [comment]
Mostly just a text oddity, I think - in other sections of MekHQ they are listed as "Battle Armor Infantry" or "Battle Armor Pilots". Not necessarily the most urgent issue to fix.

**Comment by IllianiCBT** [comment]
I believe this was fixed earlier in the year. I'm going to go ahead and close this, but I'll tag you, @psikomonkie, as I believe you're the one who did the BA fixes. Can you reopen this case if that didn't resolve this. Ta. :)
